app/javascript/components/forms/DeleteUserForm.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import {
+  Button, Stack,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import Form from './Form';
+import useDeleteUser from '../../hooks/mutations/users/useDeleteUser';
+import { useAuth } from '../../contexts/auth/AuthProvider';
+import Spinner from '../shared/stylings/Spinner';
+
+export default function DeleteUserForm({ handleClose }) {
+  const currentUser = useAuth();
+  const methods = useForm();
+  const { isSubmitting } = methods.formState;
+  const { onSubmit } = useDeleteUser(currentUser?.id);
+
+  return (
+    <Form methods={methods} onSubmit={onSubmit}>
+      <Stack direction="horizontal" gap={1}>
+        <Button variant="secondary" onClick={handleClose}>
+          Close
+        </Button>
+        <Button variant="danger" type="submit" disabled={isSubmitting}>
+          Delete
+          { isSubmitting && <Spinner /> }
+        </Button>
+      </Stack>
+    </Form>
+  );
+}
+
+DeleteUserForm.propTypes = {
+  handleClose: PropTypes.func.isRequired,
```

Comment:
I think this should change to `handleColse: PropTypes.func` because you are creating the `DeleteUserForm` first without the `handleClose` prop and inject afterwards with the `cloneElement` **API**.
Check this warning the browser console:
```
Warning: Failed prop type: The prop `handleClose` is marked as required in `DeleteUserForm`, but its value is `undefined`.
```

Reply:
Agreed.
But now it is complaining that it has no default value.
Do you think this is acceptable in this case?
```
DeleteUserForm.defaultProps = {
  handleClose: '',
};
```

Reply:
Hmm, setting it to an empty string is wrong it will raise the fallowing error in the browser console:
```
Warning: Failed prop type: Invalid prop `handleClose` of type `string` supplied to `DeleteUserForm`, expected `function`.
```
Try this instead:
```
DeleteUserForm.defaultProps = {
  handleClose: () => { },
};
```